editoast/src/views/work_schedules.rs

```diff
@@ -275,6 +295,238 @@ async fn project_path(
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_name = work_schedule_group_name.unwrap_or(Uuid::new_v4().to_string());
+
+    // Create the work_schedule_group
+    let work_schedule_group = WorkScheduleGroup::changeset()
+        .name(group_name.clone())
+        .creation_date(Utc::now())
+        .create(conn)
+        .await;
+    let work_schedule_group = work_schedule_group.map_err(|e| map_diesel_error(e, group_name))?;
+    Ok(Json(WorkScheduleGroupCreateResponse {
+        work_schedule_group_id: work_schedule_group.id,
+    }))
+}
+
+#[utoipa::path(
+    delete, path = "",
+    tag = "work_schedules",
+    params(WorkScheduleGroupIdParam),
+    responses(
+        (status = 204, description = "The work schedule group has been deleted"),
+        (status = 404, description = "The work schedule group does not exist"),
+    )
+)]
+async fn delete_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    raw_group_id: Path<WorkScheduleGroupIdParam>,
+) -> Result<impl IntoResponse> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_id = raw_group_id.id;
+
+    conn.transaction(|conn| {
+        Box::pin(async move {
+            // Check that the group exists
+            let work_schedule_group =
+                WorkScheduleGroup::retrieve_or_fail(&mut conn.clone(), group_id, || {
+                    WorkScheduleError::WorkScheduleGroupNotFound { id: group_id }
+                })
+                .await?;
+
+            // List work schedules to be deleted
+            let selection_setting = SelectionSettings::new()
+                .filter(move || WorkSchedule::WORK_SCHEDULE_GROUP_ID.eq(group_id));
+            let work_schedule_ids = WorkSchedule::list(&mut conn.clone(), selection_setting)
+                .await?
+                .into_iter()
+                .map(|work_schedule| work_schedule.id);
+
+            // Delete them
+            use crate::models::DeleteBatch;
+            let conn = &mut db_pool.get().await?;
+            WorkSchedule::delete_batch(conn, work_schedule_ids).await?;
+
+            // Delete the group itself
+            WorkScheduleGroup::delete(&work_schedule_group, conn).await?;
+
+            Ok(axum::http::StatusCode::NO_CONTENT)
+        })
+    })
+    .await
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "work_schedules",
+    responses(
+        (status = 201, body = Vec<i64>, description = "The existing work schedule group ids"),
+    )
+)]
+async fn list_groups(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+) -> Result<Json<Vec<i64>>> {
```

Comment:
Could this endpoint be paginated? Considering you're already using `models::List` that won't be a big change, you can use this as a template https://github.com/OpenRailAssociation/osrd/blob/d126f80d29dd7ab816eec1e4d0a9d8615a98be20/editoast/src/views/projects.rs#L189-L212.

I understand that this just returns a list of IDs so I'm not worried about the response size, but:

1. if we add more information to the response like you mentioned in the description, that might become necessary
2. for consistency with other listing endpoints
3. bounding requests size is a good practice (which we don't really follow 🙈, but at least we can try when it's an easy win...)

Wdyt?

Reply:
I don't *really* think it's worth the troubles for the groups, there's not supposed to be a lot of them in the db (probably ranging from 0 to 2 most of the times).

That said, when getting the content of a group it would indeed make sense, I'll change it.

Reply:
Done in dbeed01b77f8549c185e4d08073f49b1c937b14e (just for the group content)